File: springboot2_x/src/main/java/kr/deity/springboot2_x/web/BaseResponse.java
package kr.deity.springboot2_x.web;

import lombok.Data;

@Data
public class BaseResponse {
    int status;
    String error;
    String message;

    public BaseResponse(){
        this.status = 200;
        this.error = "SUCCESS";
        this.message="";
    }
    public BaseResponse(String message) {
        this.status = 500;
        this.error="FAIL";
        this.message = message;
    }
}

File: kotilWebSecurity/src/main/kotlin/kr/pe/deity/kotilwebsecurity/web/BaseResponse.kt
package kr.pe.deity.kotilwebsecurity.web

abstract class BaseResponse(
    open val status:Int?,
    open val error:String?,
    open val message:String?,
)

data class DataResponse<T>(
    val data: T?
) : BaseResponse(status = 200, error="SUCCESS", message = "")

data class ErrorResponse(
    override val message: String?,
): BaseResponse(status=500, error="FAIL", message = message)